drivers/watchdog/wdt_esp32.c

```diff
@@ -22,135 +42,120 @@ struct wdt_esp32_data {
 	u32_t timeout;
 	enum wdt_mode mode;
 	wdt_callback_t callback;
-	struct device *dev;
 };
 
-static struct wdt_esp32_data shared_data;
+struct wdt_esp32_config {
+	void (*connect_irq)(void);
+	const struct wdt_esp32_regs_t *base;
+	const struct timer_esp32_irq_regs_t irq_regs;
+
+	const struct {
+		int source;
+		int line;
+	} irq;
+};
+
+#define DEV_CFG(dev) \
+	((const struct wdt_esp32_config *const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct wdt_esp32_data *)(dev)->driver_data)
+#define DEV_BASE(dev) \
+	((volatile struct wdt_esp32_regs_t  *)(DEV_CFG(dev))->base)
 
 /* ESP32 ignores writes to any register if WDTWPROTECT doesn't contain the
  * magic value of TIMG_WDT_WKEY_VALUE.  The datasheet recommends unsealing,
  * making modifications, and sealing for every watchdog modification.
  */
-static inline void wdt_esp32_seal(void)
+static inline void wdt_esp32_seal(struct device *dev)
 {
-	volatile u32_t *reg = (u32_t *)TIMG_WDTWPROTECT_REG(1);
```

Comment:
Any reason to not use the macros from the ESP32 SDK?

Reply:
- It is more easier to manage device dependent registers using `dev` instance. 
- I found it more readable and modular to group everything related to registers addresses and offset in config struct
- I used the macros in config struct [lines 242,243](https://github.com/zephyrproject-rtos/zephyr/pull/20265/files#diff-6dc0427c13d71084aba1bc052bfbe136R242-R243)
- I'm trying to be independent of esp-idf as much as i can, so we gain more control over our code.

Correct me if i'm wrong or missed any thing

Reply:
@lpereira any feedback?

Reply:
It's kind of hard to get rid of ESP-IDF because of the HAL library, but this is fine too.